### PR TITLE
unit test for url_service

### PR DESCRIPTION
**Pull request by vladislavprovich**
Add unit test for url_service

### CODE REVIEW

File: internal/service/url_service_test.go [hunk 1]
```diff
@@ -13,7 +13,16 @@ type MockURLRepository struct {
 	mock.Mock
 }
 
-// TODO add mock methods for save and get url
+// created two mock method for save and ger url address
+func (m *MockURLRepository) SaveURL(url models.URL) error {
+	args := m.Called(url)
+	return args.Error(1)
```
Comment (vkotsiuba99) [inline internal/service/url_service_test.go]: why 1 (Error(1))

File: internal/service/url_service_test.go [hunk 2]
```diff
@@ -25,7 +34,63 @@ func (m *MockURLRepository) GetStats(shortURL string) (models.StatsResponse, err
...
+	testURL := models.URL{
+		ID:          "testone",
+		OriginalURL: shortrequesr.URL,
+		ShortURL:    shortURL,
+		CustomAlias: nil,
+		CreatedAt:   time.Now(),
+		ExpiredAt:   nil,
+	}
+	// TODO FIX THIS....
+	if shortrequesr.CustomAlias != nil && *shortrequesr.CustomAlias != "" {
```
Comment (vkotsiuba99) [inline internal/service/url_service_test.go]: it is not necessary to write if/else when you want to test several scenarios, it is better to write separate tests, if/else takes place when we have table tests with a huge number of different variations of the input data

Comment (vkotsiuba99) [reply internal/service/url_service_test.go]: fix all TODO

File: internal/service/url_service_test.go (hunk 1 above)
Comment (vladislavprovich) [reply internal/service/url_service_test.go]: after the save method, we return the second value Error(1)

File: internal/service/url_service_test.go (hunk 2 above)
Comment (vladislavprovich) [reply internal/service/url_service_test.go]: fixed

**Pull request merged**